addons/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/handler/ModbusPollerThingHandlerImpl.java

```diff
@@ -330,4 +414,32 @@ public void childHandlerDisposed(ThingHandler childHandler, Thing childThing) {
         return pollTask;
     }
 
+    /**
+     * Refresh the data
+     *
+     * If data or error was just recently received, return the cached response
+     */
+    @Override
+    public void refresh() {
+        PollTask pollTask = this.pollTask;
+        if (pollTask == null) {
+            return;
+        }
+
+        long cacheMillis = this.config.getCacheMillis();
```

Comment:
I would parse this value in `initialize` already and not on every refresh. You can also set the thing to OFFLINE, in case there is a configuration error due to a broken config.

Reply:
Configuration is actually _parsed_ to an configuration object already in `initialize`. 

Modified such that the `cacheMillis` long value is a field now, set in `initialize`. No configuration error should be possible, all long values are accepted (negative and zero disable polling).